tests/base_test.go

```diff
@@ -0,0 +1,649 @@
+		}
+
+		fileinput, err := os.ReadFile(cacertFile)
+		if err != nil {
+			t.Fatalf("Could not read %s", cacertFile)
+		}
+		cacert = fmt.Sprintf("%s", fileinput)
+
+		sc = os.Getenv("STORAGE_CLASS")
+		if sc == "" {
+			t.Fatal("STORAGE_CLASS is not set")
+		}
+
+		vmId = os.Getenv("OVIRT_VM")
+		if vmId == "" {
+			t.Fatal("OVIRT_VM is not set")
+		}
+		timeoutInput := os.Getenv("MIGRATION_TIMEOUT")
+		if timeoutInput != "" {
+			migrationTimeout = timeoutInput
+		}
+	} else {
+		username = "admin@internal"
+		password = "123456"
+		ovirtURL = "https://fakeovirt.konveyor-forklift:12346/ovirt-engine/api"
+		cacert = "-----BEGIN CERTIFICATE-----\nMIIDtzCCAp+gAwIBAgIUeHhXLiJJpaIld4tY7I5f8m7RjtwwDQYJKoZIhvcNAQEL\nBQAwQjELMAkGA1UEBhMCWFgxFTATBgNVBAcMDERlZmF1bHQgQ2l0eTEcMBoGA1UE\nCgwTRGVmYXVsdCBDb21wYW55IEx0ZDAeFw0yMjExMDEwOTQ5NDNaFw0zMjEwMjkw\nOTQ5NDNaMIGCMQswCQYDVQQGEwJYWDEVMBMGA1UEBwwMRGVmYXVsdCBDaXR5MRww\nGgYDVQQKDBNEZWZhdWx0IENvbXBhbnkgTHRkMT4wPAYDVQQDDDVmYWtlb3ZpcnQu\na29udmV5b3ItZm9ya2xpZnQsaW1hZ2Vpby5rb252ZXlvci1mb3JrbGlmdDCCASIw\nDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALffxVPN+Lin21wDbPnFc1weibfD\nmHqcXfzIMHaaHvV3MA3IXe5q3kmJlt1zc1ds4ydoaYRcJeNpuA7Fr4PdATHVqxZB\njC9m+wB/A16BgT69VPmnDWuzmkkn+NM8JdBRv+xOFJs1P9ndMFtHgSK1o/+sGpGj\naVuuZbOd4drG8bhYWKWhxHTDkFjGe9C82G6XY1Qk2gB2f5EA30uo/sSLYur69HiK\nHSs7r+RAaES7wlM7Nz4Hrf4XAI2qcdHNI1T5ZpDXkZNpgC6l/9m5FIUjeVzHqf0g\nMzCIPTBwjp4zJOMIQ21D4cXm2Zi/nlrM60IfAd7DZX2WqBCaUjELDHFTFUECAwEA\nAaNkMGIwQQYDVR0RBDowOIIbZmFrZW92aXJ0LmtvbnZleW9yLWZvcmtsaWZ0ghlp\nbWFnZWlvLmtvbnZleW9yLWZvcmtsaWZ0MB0GA1UdDgQWBBSoSoPKSAo3U0bDo/0U\nhgAceY4ZgTANBgkqhkiG9w0BAQsFAAOCAQEAB4B7DCipsCxRYq8t0+a1IUYCcWbs\nAI40GhRg2k6/p4gP2WU/v6TRpQ3HuxY2Tty3IdezMBZzeW2NgQBxt/Eo9cfWTs/5\nyX7kICzQvrOabTgF84EcCeRlP+It9xjfWWG4adEyv0XETjYUG85rBW7ud8n6dOMY\ncOgdOcJ1VvqDAfAF1uDYOIpEdqESrnqTxj+qewlBpBv2Y8m8na+AL9Sy6rdkV90e\nxKfiJG/OQNJEzgpK1XUSh/Eg3gEnrpuA+jWxShOL0zJFbMQDA0oxv6wnBxLIZXpQ\n8dDsiipGN6HDZUpzxHGFDDPB7xtCFGjyLBXCLegh+31XnZ7w6AzY3mIBYw==\n-----END CERTIFICATE-----"
+		sc = "standard"
+		vmId = "31573c08-717b-43e0-825f-69a36fb0e1a1"
+	}
+
+	// Register
+	err = v1beta1.SchemeBuilder.AddToScheme(scheme.Scheme)
+	if err != nil {
+		t.Error(err, "Failed to build scheme")
+	}
+	err = apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		t.Error(err, "Failed to add scheme")
+	}
+
+	cl, err := client.New(conf, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		t.Error(err, "Failed to create client")
+	}
+
+	namespaceName := fmt.Sprintf("provider-test-ns-%d", time.Now().Unix())
+	var localProviderNamespace string
+	if customEnv {
+		localProviderNamespace = namespaceName
+	} else {
+		localProviderNamespace = "konveyor-forklift"
+	}
+	log.Info("Creating namespace... " + namespaceName)
+
+	namespace := &corev1.Namespace{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Namespace",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+
+	defer cl.Delete(context.TODO(), namespace, &client.DeleteOptions{})
```

Comment:
it seems like it's a preparation for running several tests on the same k8s instance but we create the instance per test, right? it's not the cleanup part that bothers me but the fact we're diverging from a typical flow in which providers for instance are created on the same namepsace that the controller and other forklift components run on

Reply:
the initial thought was - we should create a vm, migrate and then cleanup everything. currently we do create instance per test, so it's fine to be removed. later on, we may think of re-usage and therefore some cleanup.

Reply:
I see, so let's take it step by step then - if we don't need that extra namespace now, we can use the namespace of the forklift components. when we get to the point we execute multiple tests on the same k8s instance, we can discuss if that would be the right way to go - I think the benefit of imitating what happens in real deployment would be great than removing the namespace

Reply:
i actually prefer to have the extra namespace and remove the namespace cleanup. it's easier to follow afterwards, if you wish to run locally you won't need to remove every resource, you will just remove the namespace.

Reply:
because you will know you have everything (mappings, plan, migration, vm) under the new namespace. if you wish to do another test on the environment just delete the namespace and run the test again.

Reply:
ah I see, the comma characters made it look like you list several reasons and I thought you meant that it would be easier to follow the result of the tests after their execution while what you meant was to follow up with execution the same test (or other test that may conflict with the previous one) on the same environment..

I see your point but I see much greater value in being aligned with how the system works in production. I think we can address the use cases you mentioned differently, e.g., if the test creates a provider then I'd expect that all those entities would be removed when removing the provider - that way we exercise real-world scenarios rather than removing a namespace of plans/migrations/mapping/etc - a scenario that shouldn't really happen..